engine/src/flutter/testing/dart/high_bitrate_texture_test.dart

```diff
@@ -0,0 +1,235 @@
+    expect(data.getUint8(offset), 255, reason: 'Center Red');
+    expect(data.getUint8(offset + 1), 255, reason: 'Center Green');
+    expect(data.getUint8(offset + 2), 255, reason: 'Center Blue');
+    expect(data.getUint8(offset + 3), 255, reason: 'Center Alpha');
+
+    image.dispose();
+  });
+}
+
+Future<Image> _drawWithShader(Image image) async {
+  final FragmentProgram program = await FragmentProgram.fromAsset('sdf.frag.iplr');
+  final FragmentShader shader = program.fragmentShader();
+  shader.setFloat(0, image.width.toDouble());
+  shader.setFloat(1, image.height.toDouble());
+  shader.setImageSampler(0, image);
+
+  final recorder = PictureRecorder();
+  final canvas = Canvas(recorder);
+  canvas.drawRect(
+    Rect.fromLTWH(0, 0, image.width.toDouble(), image.height.toDouble()),
+    Paint()..shader = shader,
+  );
+  final Picture picture = recorder.endRecording();
+  return picture.toImageSync(image.width, image.height);
+}
+
+Future<Image> _drawWithCircleShader(int width, int height, TargetPixelFormat format) async {
+  final FragmentProgram program = await FragmentProgram.fromAsset('circle_sdf.frag.iplr');
+  final FragmentShader shader = program.fragmentShader();
+  shader.setFloat(0, width.toDouble());
+  shader.setFloat(1, height.toDouble());
+
+  final recorder = PictureRecorder();
+  final canvas = Canvas(recorder);
+  canvas.drawRect(
+    Rect.fromLTWH(0, 0, width.toDouble(), height.toDouble()),
+    Paint()..shader = shader,
+  );
+  final Picture picture = recorder.endRecording();
+  return picture.toImageSync(width, height, targetFormat: format);
+}
+
+Future<Image> _drawIntoImage(Image image) {
+  final recorder = PictureRecorder();
+  final canvas = Canvas(recorder);
+  canvas.drawImage(image, Offset.zero, Paint());
+  final Picture picture = recorder.endRecording();
+  return picture.toImage(image.width, image.height);
+}
+
+Future<Image> _createRGBA32FloatImage(int width, int height) async {
+  final double radius = width / 4.0;
+  final floats = List<double>.filled(width * height * 4, 0.0);
+  for (var i = 0; i < height; ++i) {
+    for (var j = 0; j < width; ++j) {
+      double x = j.toDouble();
+      double y = i.toDouble();
+      x -= width / 2.0;
+      y -= height / 2.0;
+      final double length = math.sqrt(x * x + y * y);
+      final int idx = i * width * 4 + j * 4;
+      floats[idx + 0] = length - radius;
+      floats[idx + 1] = 0.0;
+      floats[idx + 2] = 0.0;
+      floats[idx + 3] = 1.0;
+    }
+  }
+  final floatList = Float32List.fromList(floats);
+  final intList = Uint8List.view(floatList.buffer);
+  final completer = Completer<Image>();
+  decodeImageFromPixels(
+    intList,
+    width,
+    height,
+    PixelFormat.rgbaFloat32,
+    targetFormat: TargetPixelFormat.rgbaFloat32,
+    (Image image) {
+      completer.complete(image);
+    },
+  );
+  return completer.future;
+}
+
+Future<Image> _createR32FloatImage(int width, int height) async {
```

Comment:
A comment on the type of image it creates (i.e. a circle/ellipse half the width/height).